bindings/python/test/test_mongocrypt.py

```diff
@@ -134,6 +193,36 @@ def test_mongocrypt_validation(self):
                 MongoCryptError, "local key must be 96 bytes"):
             MongoCrypt(invalid_key_len_opts, callback)
 
+    @unittest.skipUnless(not PY3, 'bytes are encoded to BSON binary on Py 3')
+    def test_setopt_local_kms_py2(self):
+        callback = MockCallback()
+        kms_dict = {'local': {'key': 'foo'}}
+
+        # Case 1: pass key as string containing bytes (valid)
+        kms_dict['local']['key'] = '\x00' * 96
+        options = MongoCryptOptions(kms_dict)
+        MongoCrypt(options, callback)
```

Comment:
It's preferable to call MongoCrypt.close() explicitly rather than relying on `__del__`. One reason is described here: https://cffi.readthedocs.io/en/latest/using.html?highlight=__del__#memory-pressure-pypy :
```
MongoCrypt(options, callback).close()
# or 
mc = MongoCrypt(options, callback)
mc.close()
```

I added `__del__` as a additional layer of safety but I don't think we should get into the habit of relying on it in practice.